pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy

```diff
@@ -266,4 +253,11 @@ public class Environment implements Serializable {
         public boolean isLiteral
         public Object value
     }
+
+    private String resolveAsScript(Binding binding, String script) {
+        String toRun = Utils.prepareForEvalToString(script)
+        SecureGroovyScript toExec = new SecureGroovyScript(toRun, true)
+            .configuring(ApprovalContext.create().withCurrentUser().withKey(Utils.getCurrentJobName()))
```

Comment:
So I think you should remove the usage of `whithKey` then to avoid future confusion, since it's not used anyway.